app/models/deferred_queue.rb

```diff
@@ -5,8 +5,12 @@ def initialize
 
   def commit
     @jobs.each do |worker, custom_queue_name, args|
-      worker.set(queue: custom_queue_name) if custom_queue_name.present?
-      worker.perform_async(*args)
+      if custom_queue_name
```

Comment:
Not blocking at all
A quick read of the source suggests you could dry this up a little by using the duck typed `Sidekiq::Worker::Setter` class
https://github.com/mperham/sidekiq/blob/903448a9be80b4bbf6801b1a95f7399a9aa3dcad/lib/sidekiq/worker.rb#L196
https://github.com/mperham/sidekiq/blob/903448a9be80b4bbf6801b1a95f7399a9aa3dcad/lib/sidekiq/worker.rb#L163
```
if custom_queue_name
  worker = worker.set(queue: custom_queue_name)
end
worker.perform_async(*args)
``